null_int.go

```diff
@@ -0,0 +1,103 @@
+package dosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// NullInt is a type that holds an optional int64 value.
```

Comment:
I didn't want to take dependency on the whole sql package just to bring in the Null types. I can add Scan() support if you'd like.